main.tf

```diff
@@ -0,0 +1,362 @@
+##########################################################################################
+########################################   S3  ###########################################
+##########################################################################################
+#create JSON bucket
+resource "aws_s3_bucket" "json-bucket" {
+  bucket = var.json_bucket_name
+}
+
+#create S3 bucket for csv objects
+resource "aws_s3_bucket" "csv-bucket" {
+  bucket = var.csv_bucket_name
+}
+
+#JSON bucket bucket-policy
+data "aws_iam_policy_document" "lambda_s3_get_permissions" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
+      "s3:ListAllMyBuckets",
+    ]
+    resources = ["${aws_s3_bucket.json-bucket.arn}/*"]
+  }
+}
+
+resource "aws_iam_policy" "lambda_s3_permissions_to_get_from_s3" {
+  name        = "lambda_s3_get_permissions"
+  path        = "/"
+  description = "IAM policy for get from s3"
+  policy      = data.aws_iam_policy_document.lambda_s3_get_permissions.json
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_s3_get" {
+  role       = aws_iam_role.iam_for_lambda.name
+  policy_arn = aws_iam_policy.lambda_s3_permissions_to_get_from_s3.arn
+}
+
+#CSV bucket bucket-policy
+data "aws_iam_policy_document" "lambda_s3_put_permissions_document" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:PutObject",
+      "s3:ListAllMyBuckets",
+    ]
+    resources = ["${aws_s3_bucket.csv-bucket.arn}/*"]
+  }
+}
+
+resource "aws_iam_policy" "lambda_s3_put_permissions" {
+  name        = "lambda_s3_put_permissions"
+  path        = "/"
+  description = "IAM policy for put to s3"
+  policy      = data.aws_iam_policy_document.lambda_s3_put_permissions_document.json
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_s3" {
+  role       = aws_iam_role.iam_for_lambda.name
+  policy_arn = aws_iam_policy.lambda_s3_put_permissions.arn
+}
+
+data "aws_iam_policy_document" "allow_access_from_lambda_fn_document" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = [
+      "s3:GetObject",
+      "s3:ListBucket",
+      "s3:*",
+    ]
+
+    resources = [
+      aws_s3_bucket.json-bucket.arn,
+      "${aws_s3_bucket.json-bucket.arn}/*",
+    ]
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = [aws_lambda_function.csv_to_json_lambda.arn]
+    }
+  }
+}
```

Comment:
I believe this bucket policy is not needed, since the s3 notifications are sent to SQS and not lambda. To read/write data to s3, lambda will use the role permissions